v1/plugin/plugin.go

```diff
@@ -200,14 +267,10 @@ func applySecurityArgsToMeta(m *meta, args *Arg) error {
 
 // buildGRPCServer configures and builds GRPC server ready to server a plugin
 // instance
-func buildGRPCServer(typeOfPlugin pluginType, name string, version int, opts ...MetaOpt) (server *grpc.Server, m *meta, err error) {
-	args, err := getArgs()
-	if err != nil {
-		return nil, nil, err
-	}
+func buildGRPCServer(typeOfPlugin pluginType, name string, version int, arg *Arg, opts ...MetaOpt) (server *grpc.Server, m *meta, err error) {
 	m = newMeta(typeOfPlugin, name, version, opts...)
 
-	if err := applySecurityArgsToMeta(m, args); err != nil {
+	if err := applySecurityArgsToMeta(m, arg); err != nil {
```

Comment:
We don't return arguments but rather an argument which is the valid JSON argument that is deserialized and returned.  I think I prefer to call the function getArg() where it's return value get's assigned to arg.